### PR TITLE
Modify resource constraints for master components.

### PR DESCRIPTION
**Pull request by mborsz**
The growth is caused by increasing --kube-api-qps in kube-scheduler.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by shyamjvs** [comment]
@mborsz - Have you verified the cause to be increased qps? (I'm quite sure it's that, but better to confirm)

**Comment by mborsz** [comment]
Logs of kube-scheduler contain values of all flags. I just checked few runs before the spike and few after and indeed, as you suggested the value for kube-api-qps has changed.

**Comment by shyamjvs** [comment]
Could you also bump the threshold for controller-manager, which is also causing flakes.. e.g:

- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/15415
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/15424

**Comment by mborsz** [comment]
Done. PTAL

File: test/e2e/scalability/density.go [hunk 1]
```diff
@@ -171,9 +171,9 @@ func density30AddonResourceVerifier(numNodes int) map[string]framework.ResourceC
 		}
 	} else {
 		if numNodes <= 100 {
-			apiserverCPU = 1.8
+			apiserverCPU = 2.2
 			apiserverMem = 1700 * (1024 * 1024)
-			controllerCPU = 0.6
+			controllerCPU = 0.9
```
Comment (shyamjvs) [inline test/e2e/scalability/density.go]: I think 0.8 would be better (0.9 is giving more slack than required) - unless there's some counter example

**Comment by shyamjvs** [comment]
One comment. Also please update the PR title.

File: test/e2e/scalability/density.go (hunk 1 above)
Comment (mborsz) [reply test/e2e/scalability/density.go]: Fixed.

**Comment by mborsz** [comment]
PTAL

**Comment by shyamjvs** [comment]
@mborsz These flakes are also happening on our [1.11 job](https://k8s-testgrid.appspot.com/sig-scalability-gce#gce-1.11) and we need to eliminate those. I guess we have 2 options:

- cherry-pick this PR to 1.11
- bump down qps values for that job back to defaults

I'm in favour of the latter to avoid a cherrypick (also 1.11 is not where we aimed for higher throughput anyway).

**Comment by shyamjvs** [comment]
To add some context, we've recently switched all our scalability jobs (even smallest clusters, i.e 100-node ones) to use 100 qps for scheduler and controller-manager. Earlier we were doing this only for large clusters (i.e O(1k)-nodes) which had bigger masters. For smaller masters, this can become a considerable problem as we may likely cause CPU starvation (for e.g we have only 4-core master for 100-node cluster). See how CPU usage for master components increased:

apiserver (increased by ~0.6 cores):

100-apiserver

controller-manager (increase by 0.2 - 0.3 cores):

100-cm

etcd (increase by ~0.2 cores):

100-etcd

Interestingly for scheduler, the CPU usage fell by ~0.1 cores (need to understand this):

100-sched

Overall, there seems to be a ~1 core increase in usage. So we should be very careful with trying to increase throughput for smaller clusters.

cc @wojtek-t @bsalamat @davidopp @gmarek

**Pull request merged**